### PR TITLE
Return error and exit on invalid command

### LINKED ISSUE #809: Lerna should error when provided an invalid command
```
$ lerna boostrap
```

## Expected Behavior

```
lerna: invalid command "boostrap"
```
Exit code non-zero.

## Current Behavior

No output. Exit code zero.

## Context
We had a build script with a lerna command misspelled that took us way too long to diagnose due to lerna quietly exiting.

## Your Environment

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 2.0.0-rc.2 |
| `npm --version`  | 4.4.4 |
| `yarn --version` | 0.23.4 |
| `node --version` | v6.2.2 |

| OS | Version |
| --- | --- |
| Windows 10 Enterprise | 14393 |

### PR DESCRIPTION
**Pull request by Siilwyn**
Return error message and exit with non-zero code. Disable help on fail
so that the error is clearly visible.

Resolves #809

## Motivation and Context

See issue #809

## How Has This Been Tested?

It's not, only seeing command specific tests. If I should add a test please point to where it should be added.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/cli.js [hunk 1]
```diff
@@ -39,6 +39,8 @@ export default function CLI(argv, cwd) {
     .help("h").alias("h", "help")
     .version().alias("v", "version")
     .wrap(cli.terminalWidth())
+    .strict()
+    .showHelpOnFail(false)
```
Comment (evocateur) [inline src/cli.js]: We should provide a message to `showHelpOnFail()`:
```js
    .showHelpOnFail(false, "Pass --help to see all available commands and options")
```

The existing `demandCommand()` usage should also include a message, to account for cases when `lerna` is called without arguments.
```js
    .demandCommand(1, "A command is required")
```

Comment (Siilwyn) [inline src/cli.js]: Thanks for the feedback!
It even returns `A command is required.` when given an incorrect argument like `lerna nope`. But it seems `yargs` has no way of preventing that.

Comment (evocateur) [inline src/cli.js]: Messing around locally, without the extra `"A command is required"` message to `demandCommand()`, I was getting less-than-useful feedback like `Unknown argument: nope`.

Comment (Siilwyn) [inline src/cli.js]: So we're good to go right?

The CI is doing some funky stuff though.

Comment (evocateur) [inline src/cli.js]: Yes, thank you!

**Pull request merged**